crates/storage/src/pack.rs

```diff
@@ -117,6 +117,14 @@ where
         }
     }
 
+    /// Creates a new packed value.
+    pub fn new_with_key(value: T, key: Key) -> Self {
+        Self {
+            inner: value,
+            key: Some(key),
+        }
+    }
```

Comment:
Still super low-level API. Please think twice if you really need something like this and if you do think twice again if it really should be made public or even gated with something a la `#[cfg(test)]`.

Reply:
It is not much better than the previous `set_key` method from an API perspective. Keep in mind that the `Pack` data structure is a high-level, user targeting data structure.

Reply:
So I've made the method non-public. But please not: this is no test feature, hence we can't gate it. This method is called from `Pack::SpreadLayout::pull_spread` and used to set the `key` behind which this object is on `Pack.key`. This key is needed for `Drop` to know what to clear.